capactio/capact
# Rename Tag to Attribute

**Description**

Changes proposed in this pull request:

- Rename Tag entity to Attribute

### Scope

The following places were checked for “Tag” occurrences and renamed, if there were at least one occurrence (in proper context of our entity):

- [This repository](https://github.com/Project-Voltron/go-voltron) (excluding previous investigations)
- [OCF Draft](https://docs.google.com/document/d/1ud7xL3bXxEXtVPE8daA_DHYacKHMkn_jx6s7eaVT-NA/edit#)
- [Voltron Architecture](https://docs.google.com/document/d/1LRwS_r_IyObtVVnWssGq9MdgFeV-GXZ_r3U3uetzt9E/edit#heading=h.olrxskfvpv4t)
- [Voltron MVP Plan](https://docs.google.com/document/d/1PWaUqH4zq_p7OnoL665z7QVI4uCcB1xllxExPcKLDfA/edit#heading=h.kjxelh634vgl) - however it is already a bit out of sync with Jira. I think it would be the best to treat Jira as the single source of truth. In this case, we need to eventually move some missing parts (maybe some “After GA” ideas) from this document to Jira.
- [Jira](https://cshark.atlassian.net/jira/software/c/projects/SV/issues/SV-152?jql=project%20%3D%20%22SV%22%20AND%20text%20~%20%22tag%22%20%20AND%20status%20IN%20%28%2210022%22%2C%20%223%22%2C%20%2210063%22%2C%20%2210106%22%29%20ORDER%20BY%20created%20DESC) - current and future tasks (previous tasks left intact): 
Places which were NOT updated:

### Out of scope

- https://github.com/Project-Voltron/scratchpad - as it contains some early drafts of OCF spec, which weren't updated for months.


### Testing

Most of the places are covered by CI checks. There is one exception: mocked OCH.
To see mocked OCH in action with new attributes:

1. Run:

    ```bash
    MOCK_OCH_GRAPHQL=true make dev-cluster
    ``` 

1. Navigate to https://gateway.voltron.local/graphql
2. Run the following queries:

```graphql
query TypeInstances {
  typeInstances {
    metadata {
      id
      attributes {
        path
        revision
      }
    }
  }
}

query Types {
  types {
    revisions {
      metadata {
        name
        attributes {
          metadata {
            name
            path
            prefix
            displayName
            description
            documentationURL
            supportURL
            iconURL
          }
          revision
        }
      }
    }
  }
}

query Attributes {
  attributes {
    name
    path
    prefix
    latestRevision {
      metadata {
        name
        prefix
        path
        displayName
        description
        maintainers {
          name
          email
          url
        }
        documentationURL
        supportURL
        iconURL
      }
      revision
      spec {
        additionalRefs
      }
      signature {
        och
      }
    }
  }
}

query Attribute {
  attribute(path: "cap.core.attribute.system.baseline") {
    name
    path
    prefix
    latestRevision {
      metadata {
        name
        prefix
        path
        displayName
        description
        maintainers {
          name
          email
          url
        }
        documentationURL
        supportURL
        iconURL
      }
      revision
      spec {
        additionalRefs
      }
      signature {
        och
      }
    }
  }
}

# Mock Implementations doesn't have any attributes, but if you want to test it anyway, here's the query :D
query Implementations {
  implementations {
    revisions {
      metadata {
        name
        attributes {
          metadata {
            name
          }
        }
      }
    }
  }
}

```
